File: mesh-suite-frontend/src/api/accountsPayable.ts
import { apiClient } from './client'

export type AccountsPayableStatus = 'OPEN' | 'PAID'

export interface AccountsPayable {
  id: string
  number: number
  installmentNumber: number
  totalInstallments: number
  supplierId: string
  supplierName: string
  amount: number
  issueDate: string
  dueDate: string
  paymentDate: string | null
  status: AccountsPayableStatus
  referenceId: string | null
  createdAt: string
}

export interface Page<T> {
  content: T[]
  totalElements: number
  totalPages: number
  number: number
  size: number
}

export interface ListAccountsPayableParams {
  status?: AccountsPayableStatus
  page?: number
  size?: number
  sort?: string
}

export async function listAccountsPayable(params: ListAccountsPayableParams): Promise<Page<AccountsPayable>> {
  const { data } = await apiClient.get<Page<AccountsPayable>>('/accounts-payable', { params })
  return data
}

export async function updateAccountsPayableStatus(id: string, status: AccountsPayableStatus): Promise<void> {
  await apiClient.patch(`/accounts-payable/${id}/status`, { status })
}

File: mesh-suite-backend/src/main/java/com/meshsuite/payable/domain/AccountsPayable.java
package com.meshsuite.payable.domain;

import com.meshsuite.parceiro.domain.Parceiro;
import com.meshsuite.payable.domain.enums.AccountsPayableStatus;
import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.Instant;
import java.time.LocalDate;
import java.util.UUID;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;

@Entity
@Table(name = "accounts_payable")
@Getter
@Setter
public class AccountsPayable {

    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;

    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;

    @Column(nullable = false)
    private Integer number;

    @Column(name = "installment_number", nullable = false)
    private Integer installmentNumber;

    @Column(name = "total_installments", nullable = false)
    private Integer totalInstallments;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "supplier_id", nullable = false)
    private Parceiro supplier;

    @Column(nullable = false, precision = 12, scale = 2)
    private BigDecimal amount;

    @Column(name = "issue_date", nullable = false)
    private LocalDate issueDate = LocalDate.now();

    @Column(name = "due_date", nullable = false)
    private LocalDate dueDate;

    @Column(name = "payment_date")
    private LocalDate paymentDate;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private AccountsPayableStatus status = AccountsPayableStatus.OPEN;

    @Column(name = "reference_id")
    private UUID referenceId;

    @Column(name = "created_at", nullable = false, updatable = false)
    private Instant createdAt = Instant.now();
}
